volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -732,9 +732,10 @@ def is_valid(self) -> bool:
 
         return True
 
+    @functools.lru_cache
     def add_process_layer(
         self, config_prefix: Optional[str] = None, preferred_name: Optional[str] = None
-    ):
+    ) -> str:
```

Comment:
Just want to check why this one is `str` and the other is `optional[str]`?  I imagine one throws an exception and the other doesn't.  We probably want to be consistent on that, but this PR isn't the place to bring that about...